### DISCUSSION TITLE
Is it bug ?

### DISCUSSION THREAD

**Discussion by KaungZawHtet** [microsoft/vscode-cpptools#10192] [Q&A]
OS : Mac Ventura
VS Code Version : 1.73.1
C++ extension version : 1.13.4
Hardware : M1 Pro, RAM 16 GB

Code analysis is mostly unresponsive. Even though I change the code, it can't even catch up the latest code update. This is the most notable example when I write some GRPC things. I added `; ` into the code but still showing the outdate error. It disappeared only after the cmd `Developer: Reload Window`. As you see, this screenshot is so much meme.

<img width="670" alt="Screenshot 2022-11-27 at 7 54 33 PM" src="https://user-images.githubusercontent.com/15225530/204139609-dfe50ab7-9de6-4b91-956e-fdea716d8b60.png">

**Comment by sean-mcmanus** [comment]
The flame icon in the status bar should show the status, although it's possible that "Updating IntelliSense" could block that from appearing as well. Do you see a "Running clang-tidy" status indicator when you hover over the flame icon? Are you able to click it and choose cancel and then manually re-run code analysis to get it to update? Also, if you can check if clang-tidy is running (as a process) then that would indicate if it's doing work or if there's an internal issue causing it not to run when it should.

Are using the C_Cpp.codeAnalysis.runAutomatically and C_Cpp.codeAnalysis.clangTidy.enabled both set to true or are you manually invoking it?

**Comment by KaungZawHtet** [comment]
flame icon? i dont' find any flame icon in status bar. And yes, I enabled both what u mentioned
 
<img width="596" alt="Screenshot 2022-11-29 at 3 53 27 PM" src="https://user-images.githubusercontent.com/15225530/204490172-add5ee60-85d8-46d3-a623-967cd14738b0.png">

**Reply by sean-mcmanus** [reply]
Does your status bar have space left? It should look something like
![image](https://user-images.githubusercontent.com/19859882/204629796-b1e8daa1-b8c6-4567-8442-c2199b0a3294.png)

Does the flame appear if you manually invoke code analysis after reloading the window?

**Reply by KaungZawHtet** [reply]
invoke ? how to invoke code analysis manually? I only clicked those 2 options u mentioned

Here is my status bar :
<img width="1309" alt="Screenshot 2022-11-30 at 11 38 57 AM" src="https://user-images.githubusercontent.com/15225530/204712596-485a0b63-59d1-40b7-be15-eb89e146df9e.png">

**Reply by KaungZawHtet** [reply]
Now I found flame icon. The icon appears only when I invoke code analysis manually from cmd. It's not showing for normal time. Btw, is that unresponsiveness really related to the code analysis ? Or related to intellisense ?

**Reply by sean-mcmanus** [reply]
The RunAutomatically feature normally will invoke code analysis after a file is saved after an edit. If that doesn't cause the flame icon to appear, there could be some bug we may need to investigate, i.e. you could file an issue on it. If IntelliSense is blocking code analysis you would see "Updating IntelliSense" when hovering over the flame icon in addition to the code analysis message.

**Reply by KaungZawHtet** [reply]
Flame Icon appears after saving file and disappears again. If so, it's working as u said. Let me check one more time by knowing how code analyzer works. After that, I will tell u back

**Comment by KaungZawHtet** [comment]
This is `c_cpp_properties.json` file

```json
{
    "configurations": [
        {
            "name": "Mac",
            "includePath": [
                "${workspaceFolder}/**",
                "${vcpkgRoot}/arm64-osx/include"
            ],
            "defines": [],
            "macFrameworkPath": [
                "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks"
            ],
            "compilerPath": "/usr/bin/clang",
            "cStandard": "c17",
            "cppStandard": "c++17",
            "intelliSenseMode": "macos-clang-arm64",
            "configurationProvider": "ms-vscode.cmake-tools"
        }
    ],
    "version": 4
}
```
